ocaml/xapi/xapi_sr.ml

```diff
@@ -387,15 +387,15 @@ let update_vdis ~__context ~sr db_vdis vdi_infos =
       | loc, None, Some v -> Some v
       | _, _, _ -> None
     ) db_vdi_map scan_vdi_map in
-  let to_update = StringMap.merge (fun loc db scan -> match loc, db, scan with
-      | loc, Some (r, v), Some vi -> Some (r, v, vi)
-      | _, _, _ -> None
-    ) db_vdi_map scan_vdi_map in
 
   let find_vdi db_vdi_map loc =
     if StringMap.mem loc db_vdi_map
     then fst (StringMap.find loc db_vdi_map)
-    else Ref.null in
+    else
+      (* CA-254515: Also check for the snapshoted VDI in the database *)
+      try Db.VDI.get_by_uuid ~__context ~uuid:loc
```

Comment:
So are there some minimal improvements we could make here? I'd rather not start changing too much of the logic because, as you said, `find_vdi` is only used for finding `snapshot_of` fields, which AFAIK is not a concern for ISOs, and we may end up over-optimizing when there is actually no issue...